prov/mrail/src/mrail_init.c

```diff
@@ -165,7 +165,7 @@ int mrail_get_core_info(uint32_t version, const char *node, const char *service,
 		FI_DBG(&mrail_prov, FI_LOG_CORE,
 		       "--- Begin fi_getinfo for rail: %zd ---\n", i);
 
-		ret = fi_getinfo(version, NULL, NULL, 0, core_hints, &info);
+		ret = fi_getinfo(version, NULL, NULL, OFI_GETINFO_INTERNAL, core_hints, &info);
```

Comment:
I think these changes should be pulled into a separate patch.

Along this line, what prevents mrail from recursively calling getinfo to itself?  The CORE_PROV_ONLY flag is used by the util providers, but that's not used by mrail.

Reply:
CORE_PROV_ONLY can't be used here because mrail can be layered over other utility providers. Recursion is not going to happen because "^ofi_mrail" is explicitly added to core_hints->fabric_attr->prov_name.

This part is included here to get mrail layering to work (otherwise the intermediate result is filtered out).

Reply:
I understood why CORE_PROV flag couldn't be used.  I forgot that we have the ^mrail option to avoid recursion